gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -249,13 +258,21 @@ public Completable saveFeature(Feature feature) {
         .andThen(dataSyncWorkManager.enqueueSyncWorker(feature.getId()));
   }
 
+  private Maybe<Project> loadLastActiveProject() {
+    if (isOfflineModeEnabled()) {
```

Comment:
This and `loadProjects()` would need to be updated to be lazy, i.e., logic only evaluate on subscribe. This can be achieved by returning a single stream as we do for `loadRecords`:

https://github.com/google/ground-android/blob/55b3ebcb358b6e51d7b2741a11033d4f3c213ed4/gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java#L170-L179

In that case we rely on the remote store returning an error immediately when not available. In that case we fall back to the local data store. In this case, the logic might be:

1. Try to sync remote project config to local db
2. Load config from local db.

Reply:
See this comment for example of the logic for falling back to local db mentioned in other comments.